### PR TITLE
fix(ui): add static route form improvements

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Add static route form improvements

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to a subnet details page.
- Scroll down to the static routes table and click "Add static route".
- Enter some valid details and submit.
- There's a bug at the moment that might require you to refresh the page to see the new static route: https://bugs.launchpad.net/maas/+bug/1960462.

### CODE REVIEW

File: ui/src/app/base/components/SubnetSelect/SubnetSelect.tsx [hunk 1]
```diff
@@ -38,7 +42,11 @@ export const SubnetSelect = ({
     subnets = subnets.filter((subnet) => subnet.vlan === vlan);
   }
 
-  const subnetOptions = subnets.map((subnet) => ({
+  if (subnets && filterFunction) {
```
Comment (huwshimi) [inline ui/src/app/base/components/SubnetSelect/SubnetSelect.tsx:45]: @petermakowski it turns out this prop existed in the type, but wasn't implemented.

File: ui/src/app/subnets/views/SubnetDetails/StaticRoutes/StaticRoutes.tsx [hunk 2]
```diff
@@ -164,11 +165,11 @@ const StaticRoutes = ({ subnetId }: Props): JSX.Element | null => {
     <TitledSection
       title="Static routes"
       buttons={
-        isSuperUser ? (
+        isAdmin ? (
           <Button
             disabled={isAddStaticRouteOpen}
             onClick={() => {
-              setIsAddStaticRouteOpen(true);
+              setExpanded({ type: ExpandedType.Create });
```
Comment (huwshimi) [inline ui/src/app/subnets/views/SubnetDetails/StaticRoutes/StaticRoutes.tsx:172]: I didn't spot this in review, but it turns out this type already existed in the file.

**Review by Caleb-Ellis — APPROVED** [review]

File: ui/src/app/subnets/views/SubnetDetails/StaticRoutes/StaticRoutes.tsx [hunk 3]
```diff
@@ -151,9 +152,9 @@ const StaticRoutes = ({ subnetId }: Props): JSX.Element | null => {
   );
   const subnets = useSelector(subnetSelectors.all);
   const subnetsLoading = useSelector(subnetSelectors.loading);
-  const isSuperUser = useSelector(userSelectors.isSuperUser);
+  const isAdmin = useSelector(authSelectors.isAdmin);
```
Comment (petermakowski) [inline ui/src/app/subnets/views/SubnetDetails/StaticRoutes/StaticRoutes.tsx:155]: thanks for updating @huwshimi 

I was looking in the `user` selectors looking for priviledges as the `auth` is in the user slice.

As a side note, It feels like they should live in a single place, we're already manipulating "auth" in `user`, e.g.:
https://github.com/canonical-web-and-design/maas-ui/blob/master/ui/src/app/store/user/slice.ts#L77-L79

**Review by petermakowski — APPROVED** [review]

File: ui/src/app/base/components/SubnetSelect/SubnetSelect.test.tsx [hunk 4]
```diff
@@ -123,4 +123,25 @@ describe("SubnetSelect", () => {
...
     ]);
   });
+
+  it("can filter the subnets using a provided function", () => {
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <Formik initialValues={{ subnet: "" }} onSubmit={jest.fn()}>
+          <SubnetSelect
+            filterFunction={(subnet) => subnet.vlan === 4}
```
Comment (petermakowski) [inline ui/src/app/base/components/SubnetSelect/SubnetSelect.test.tsx:133]: Might make this test clearer and easier to understand at a glance if we changed vlan to cidr as this is what we're testing below.
```suggestion
            filterFunction={(subnet) => subnet.cidr === "172.16.2.0/24"}
```

**Pull request merged**